### PR TITLE
Grant job permission for all roles except GUEST

### PR DESCRIPTION
**Pull request by SuperYoko**
Fix job permission for roles.

### CODE REVIEW

**Comment by SuperYoko** [comment]
# 1Client Test

## 1.1 ROLES

ROLES

## 1.2 OUTPUT AFTER CHANGE FOR ADMIN JOB COMMANDS

**NOTE: only GOD can do job related before this commit**

ROOT

US1

US2

US3

US4

**Review by yixinglu — COMMENTED** [review]
Consider to design a method to test permission or user related cases.

**Comment by panda-sheep** [comment]
close #2611
`God, admin, DBA, user` has all job related permissions.
The `guest `should have the permission of `show jobs and show job xx`

File: src/graph/service/PermissionCheck.cpp [hunk 1]
```diff
@@ -115,7 +114,8 @@ Status PermissionCheck::permissionCheck(ClientSession *session,
     case Sentence::Kind::kUpdateEdge:
     case Sentence::Kind::kDeleteVertices:
     case Sentence::Kind::kDeleteTags:
-    case Sentence::Kind::kDeleteEdges: {
+    case Sentence::Kind::kDeleteEdges:
+    case Sentence::Kind::kAdminJob: {
```
Comment (SuperYoko) [inline src/graph/service/PermissionCheck.cpp:118]: @panda-sheep  Do you think it's okay if I reuse canWriteDate for kAdminJob?

Comment (panda-sheep) [reply src/graph/service/PermissionCheck.cpp:118]: reuse  `canWriteDate `for `kAdminJob`  is ok

**Comment by bright-starry-sky** [comment]
A question, for example as below:
```
create space1
create space2
create user1 
create user2
GRANT ROLE ADMIN ON space1 TO user1
GRANT ROLE ADMIN ON space2 TO user2
```

For example, the current system has two jobs of rebuild index:
```
index1 as space1
index2 as space2
```

show jobs by user1 , what results should be returned?

**Review by bright-starry-sky — APPROVED** [review]
Good job, LGTM. waiting test case .

**Comment by SuperYoko** [comment]
> A question, for example as below:
> 
> ```
> create space1
> create space2
> create user1 
> create user2
> GRANT ROLE ADMIN ON space1 TO user1
> GRANT ROLE ADMIN ON space2 TO user2
> ```
> 
> For example, the current system has two jobs of rebuild index:
> 
> ```
> index1 as space1
> index2 as space2
> ```
> 
> show jobs by user1 , what results should be returned?

I've tested this before.

Check entry listed here:
1.  Use space, if user is not granted as any roles,  space can't be used by this user.
2. "if space chosen" will be checked.
3. Permission check (for show jobs, all roles are ok)
4. The result only show current space's job.

**Review by critical27 — APPROVED** [review]
@SuperYoko will add a tck test case later in another PR

**Pull request merged**